scripts/rebuild-testbench.sh

```diff
@@ -34,22 +39,93 @@ export_CC_with_afl()
     export CC=${SOF_AFL}
 }
 
+setup_xtensa_tools_build()
+{
+    BUILD_TYPE=xt
+    BUILD_TARGET=
+    BUILD_DIR_NAME=build_xt_testbench
+
+    # check needed environment variables
+    if [[ -z "${XTENSA_TOOLS_ROOT}" ]]; then
+	echo Error: XTENSA_TOOLS_ROOT need to be set.
+	exit 1
+    fi
+
+    case $BUILD_PLATFORM in
+	tgl)
+	    compiler=xt-xcc
+	    XTENSA_TOOLS_VERSION="RG-2017.8-linux"
+	    export XTENSA_CORE="cavs2x_LX6HiFi3_2017_8"
+	    ;;
```

Comment:
I can imagine my recommendation to perform surgery and extract this from `xtensa-build-all.sh` did not get you very "excited" and that's why you used some good old copy/paste instead. Let's meet somewhere in the middle: can you please move just this `case` bit to a new `scripts/set_xtensa_params.sh` script that you source here?

Also, add a WARNING in this new file that it duplicates both `xtensa-build-zephyr.py` and `xtensa-build-all.sh`

You can pass parameters to `source`, it's not standard but it's officially supported by bash:

```bash
source "$mydir"/set_xtensa_params.sh" "$BUILD_PLATFORM"
```

As usual don't forget shellcheck :-)

Reply:
Yes, scripts/set_xtensa_params.sh is a good idea. I will do it and try to copy if possible from other script the variables names. I have an issue with MTL clang build and I had more difference in the case-esac for it  vs. tgl and needed more changes to overall source (remove -WL, EL, UNDEFINED() macros, etc.). Since it may be wrong (it ran OK though) I didn't share it in this.